### ISSUE TITLE
Deletion of files not possible.

### ISSUE THREAD

**Issue by Hardexit** [nextcloud/contacts#4126]
### Steps to reproduce

Try to delete a file

### Expected behavior

File is deleted

### Actual behavior

Error is displayed

### Server operating system

Debian

### Web server

Nginx

### Database engine version

PostgreSQL

### PHP engine version

Other

### Nextcloud version

Nextcloud Hub 8 (29.0.7)

### Updated from an older installed version or fresh install

Updated from an older version

### Signing status

```Text
No errors have been found.
```

### List of activated apps

```Text
Enabled:
  - activity: 2.21.1
  - approval: 1.3.0
  - circles: 29.0.0-dev
  - cloud_federation_api: 1.12.0
  - comments: 1.19.0
  - contacts: 6.0.0
  - contactsinteraction: 1.10.0
  - dashboard: 7.9.0
  - dav: 1.30.1
  - federatedfilesharing: 1.19.0
  - federation: 1.19.0
  - files: 2.1.1
  - files_downloadlimit: 2.0.0
  - files_pdfviewer: 2.10.0
  - files_reminders: 1.2.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - files_versions: 1.22.0
  - forms: 4.2.4
  - groupfolders: 17.0.3
  - impersonate: 1.16.0
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - maps: 1.4.0
  - nextcloud_announcements: 1.18.0
  - notifications: 2.17.0
  - oauth2: 1.17.1
  - password_policy: 1.19.0
  - photos: 2.5.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - quota_warning: 1.20.0
  - related_resources: 1.4.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - spreed: 19.0.9
  - survey_client: 1.17.0
  - systemtags: 1.19.0
  - text: 3.10.1
  - theming: 2.4.0
  - twofactor_backupcodes: 1.18.0
  - twofactor_nextcloud_notification: 3.10.0
  - twofactor_totp: 11.0.0-dev
  - twofactor_webauthn: 1.4.0
  - updatenotification: 1.19.1
  - user_ldap: 1.20.0
  - user_status: 1.9.0
  - viewer: 2.3.0
  - workflowengine: 2.11.0
Disabled:
  - admin_audit: 1.19.0
  - bruteforcesettings: 2.9.0 (installed 2.0.1)
  - encryption: 2.17.0
  - files_external: 1.21.0 (installed 0.5.2)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - firstrunwizard: 2.18.0 (installed 1.0)
  - polls: 7.2.3 (installed 7.2.3)
  - recommendations: 2.1.0 (installed 0.4.0)
  - richdocuments: 8.4.6 (installed 8.4.6)
  - support: 1.12.0 (installed 1.3.0)
  - suspicious_login: 7.0.0
  - twofactor_admin: 4.5.0 (installed 4.5.0)
  - weather_status: 1.9.0 (installed 1.0.0)
```

### Nextcloud configuration

```Text
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "mmmbox.mmmgroup.com"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "skeletondirectory": "\/home\/mmmbox_skeleton",
        "tempdirectory": "\/home\/mmmbox_tmp",
        "dbtype": "mysql",
        "version": "29.0.7.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "installed": true,
        "loglevel": 2,
        "logdateformat": "d.m.Y H:i:s",
        "logtimezone": "Europe\/Berlin",
        "log_rotate_size": 10485760,
        "maintenance": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "ldapIgnoreNamingRules": false,
        "forcessl": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "ldapUserCleanupInterval": 51,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "overwrite.cli.url": "https:\/\/mmmbox.mmmgroup.com",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\APCu",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0
        },
        "default_language": "de",
        "default_locale": "de",
        "default_phone_region": "de",
        "trashbin_retention_obligation": "auto, 60",
        "versions_retention_obligation": "auto, 120",
        "theme": "mmmbox",
        "defaultapp": "apporder",
        "simpleSignUpLink": "false",
        "activity_expire_days": "90",
        "updater.release.channel": "stable",
        "maintenance_window_start": 1
    }
}
```

### Browser

Edge

### Client operating system

Windows

### Nextcloud log

```Text
[webdav] Error: OCA\Circles\Model\ShareWrapper::getCircle(): Return value must be of type OCA\Circles\Model\Circle, null returned
	DELETE /remote.php/dav/files/USER/Software/SOFTWARE/SOFTWARE%20Win%20x64%20Test.zip
	from 212.211.191.216 by 98a4e33e-bb0c-1033-9cd7-a7755f636c69 at 11.09.2024, 10:07:54
```

**Comment by ZID-TU-Graz-Collab** [comment]
We are also affected by this issue. This issue has high priority for us

**Comment by GVodyanov** [comment]
@SebastianKrupinski Do you think that maybe it would make more sense to move this to server?

**Comment by SebastianKrupinski** [comment]
@GVodyanov no, from the error it looks like the failure is inside the circles module (so it should be moved to circles), which then makes the files delete action fail. But it is a duplicate of the attached issue.

**Comment by tron65** [comment]
We get the error as well on moving, but, as deleting is a part of moving, this is essentially the same, here is as well a full excerpt from the logs:

in this specific case, a document opened in only office can't be saved to the original place, but it is the same error as when deleting, the essence always being 

"Exception":"Exception","Message":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned in file '/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php' line 272","Code":0,"



{"reqId":"HCa4dGcS9s3ikSPmbSgw","level":3,"time":"2024-09-19T07:30:33+00:00","remoteAddr":"x.x.x.x","user":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx","app":"index","method":"POST","url":"/index.php/apps/onlyoffice/track?doc=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VySWQiOiJCMDMxMzkxMC05RkE0LURBMDEtODBGOS0wMDAzNDNGRTA2RkIiLCJvd25lcklkIjoibGx0IiwiZmlsZUlkIjoxMDY1ODAzMTM0LCJmaWxlUGF0aCI6Ii9MTFQvRG9rdW1lbnRhdGlvbi9Qcm9qZWt0ZS1Wb3JoYWJlbi9Jbi1EdXJjaGZ1ZWhydW5nLzIwMjJfVU5JVEUvW3VuaXRlX1dQMl9TSEFSRXNdL3VuaXRlX1dQMl9zaGFyZS9UMi4xX1dQX0Nvb3JkaW5hdGlvbi8wMl9NZWV0aW5ncy8yMDI0LTA5LTI0IHRvIDA5LTI2IERBUk1TVEFEVC9EYXkgMS8yMDI0LTA5LTI0IFNsb3QgMSAtIFdlbGNvbWUgYW5kIFVwZGF0ZXMucHB0eCIsInNoYXJlVG9rZW4iOm51bGwsImFjdGlvbiI6InRyYWNrIn0.uR0t_QJdK3vWs-Eidt7qT6-p4qpt1GLRj6Rix-VOsA0","message":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned in file '/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php' line 272","userAgent":"Node.js/6.13","version":"29.0.6.1","exception":{"Exception":"Exception","Message":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned in file '/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php' line 272","Code":0,"Trace":[{"file":"/var/www/app/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Route/Router.php","line":331,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/app/nextcloud/lib/base.php","line":1058,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/app/nextcloud/index.php","line":49,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/app/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","Line":170,"Previous":{"Exception":"TypeError","Message":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned","Code":0,"Trace":[{"file":"/var/www/app/nextcloud/apps/circles/lib/ShareByCircleProvider.php","line":695,"function":"getCircle","class":"OCA\\Circles\\Model\\ShareWrapper","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Share20/Manager.php","line":1688,"function":"getAccessList","class":"OCA\\Circles\\ShareByCircleProvider","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Share20/ShareHelper.php","line":51,"function":"getAccessList","class":"OC\\Share20\\Manager","type":"->"},{"file":"/var/www/app/nextcloud/apps/activity/lib/FilesHooks.php","line":598,"function":"getPathsForAccessList","class":"OC\\Share20\\ShareHelper","type":"->"},{"file":"/var/www/app/nextcloud/apps/activity/lib/FilesHooks.php","line":166,"function":"getUserPathsFromPath","class":"OCA\\Activity\\FilesHooks","type":"->"},{"file":"/var/www/app/nextcloud/apps/activity/lib/FilesHooks.php","line":108,"function":"addNotificationsForFileAction","class":"OCA\\Activity\\FilesHooks","type":"->"},{"file":"/var/www/app/nextcloud/apps/activity/lib/FilesHooksStatic.php","line":51,"function":"fileUpdate","class":"OCA\\Activity\\FilesHooks","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/legacy/OC_Hook.php","line":105,"function":"fileUpdate","class":"OCA\\Activity\\FilesHooksStatic","type":"::"},{"file":"/var/www/app/nextcloud/lib/private/Files/View.php","line":1282,"function":"emit","class":"OC_Hook","type":"::"},{"file":"/var/www/app/nextcloud/lib/private/Files/View.php","line":1214,"function":"runHooks","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Files/View.php","line":683,"function":"basicOperation","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Files/Node/File.php","line":73,"function":"file_put_contents","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/app/nextcloud/apps/onlyoffice/lib/Controller/CallbackController.php","line":588,"function":"putContent","class":"OC\\Files\\Node\\File","type":"->"},{"file":"/var/www/app/nextcloud/apps/onlyoffice/lib/Controller/CallbackController.php","line":878,"function":"OCA\\Onlyoffice\\Controller\\{closure}","class":"OCA\\Onlyoffice\\Controller\\CallbackController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/app/nextcloud/apps/onlyoffice/lib/Controller/CallbackController.php","line":587,"function":"retryOperation","class":"OCA\\Onlyoffice\\Controller\\CallbackController","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Files/Lock/LockManager.php","line":39,"function":"OCA\\Onlyoffice\\Controller\\{closure}","class":"OCA\\Onlyoffice\\Controller\\CallbackController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/app/nextcloud/apps/onlyoffice/lib/Controller/CallbackController.php","line":594,"function":"runInScope","class":"OC\\Files\\Lock\\LockManager","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":232,"function":"track","class":"OCA\\Onlyoffice\\Controller\\CallbackController","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":138,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/app/nextcloud/lib/private/Route/Router.php","line":331,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/app/nextcloud/lib/base.php","line":1058,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/app/nextcloud/index.php","line":49,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php","Line":272},"message":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned in file '/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php' line 272","exception":{},"CustomMessage":"OCA\\Circles\\Model\\ShareWrapper::getCircle(): Return value must be of type OCA\\Circles\\Model\\Circle, null returned in file '/var/www/app/nextcloud/apps/circles/lib/Model/ShareWrapper.php' line 272"}}

**Comment by ZID-TU-Graz-Collab** [comment]
Problem seems to be fixed with 29.0.9

**Comment by Hardexit** [comment]
No, we are using Nextcloud Hub 8 (29.0.9), but the error is still there
